utils/land-patterns.stanza

```diff
@@ -1071,7 +1091,154 @@ defn ipc-formula (L:Toleranced, ; the distance from edge-of-lead to edge-of-lead
+  
+    ;----------------------------------------------------------
+    ; Place the polarity marker
+    val line-width  = clearance(current-rules(), MinSilkscreenWidth)
+    val line-length = 3.0 * line-width - line-width
+    val dist  = SOLDER-MASK-REGISTRATION + SILK-SOLDER-MASK-CLEARANCE + line-width * 0.5 + 0.01
+    
+    val pad-loc = center(pad-1-loc)
+    val p1 = Point(
+      x(pad-loc) - 0.5 * x(pad-sz-ew),
+      y(pad-loc) + 0.5 * y(pad-sz-ew) + dist
+    )
+    val p2 = Point(
+      -0.5 * x(part-sz) - line-width * 0.5,
+       0.5 * y(part-sz) - line-width * 0.5,
+    )
+    val pol = max-distance(p1, p2)
+    layer(Silkscreen("pol", Top)) = Line(line-width, [
+      pol, pol + Point(line-length, 0.0)
+    ])
+
+    ;----------------------------------------------------------
+    ; Place the reference label
+    val text-y = 0.5 * (y(cy-sz) + MIN-SILKSCREEN-TEXT-HEIGHT)
+    ref-label(0.0, text-y)
+
+public defn make-quad-landpattern (num-pins:Int, 
+                                   pitch:Double, 
+                                   component-size:Toleranced, 
+                                   package-size:Toleranced, 
+                                   lead-length:Toleranced, 
+                                   lead-width:Toleranced, 
+                                   lead-type:LandProtrusionType):
+  ensure-divisible!(num-pins, 4, "make-quad-landpattern(num-pins, ...)")
+  if pitch <= 0.0:  
+    fatal("make-quad-landpattern(...) failed: pitch must be greater than 0. Called with: %_." % [pitch])
+  make-quad-landpattern(num-pins / 4, num-pins / 4, num-pins / 4, num-pins / 4, 
+                        pitch, pitch, pitch, pitch, 
+                        component-size, component-size, 
+                        package-size, package-size, 
+                        lead-length, lead-width, lead-type, DensityLevelC)
+  
+public defn make-qfn-landpattern (num-pins:Int, 
+                                  pitch:Double, 
+                                  package-size:Toleranced, 
+                                  lead-length:Toleranced, 
+                                  lead-width:Toleranced):
+  make-quad-landpattern(num-pins,
+                        pitch,
+                        package-size,
+                        package-size, 
+                        lead-length, lead-width, QuadFlatNoLeads)
+
+public defn make-qfn-landpattern (num-pins:[Int, Int, Int, Int], 
+                                  pitch:Double, 
+                                  package-size:Toleranced, 
+                                  lead-length:Toleranced, 
+                                  lead-width:Toleranced):
+  make-quad-landpattern(num-pins[0], num-pins[1], num-pins[2], num-pins[3], 
+                        pitch, pitch, pitch, pitch,
+                        package-size, package-size,
+                        package-size, package-size,
+                        lead-length, lead-width, QuadFlatNoLeads, DensityLevelC)
```

Comment:
This is incorrect. The default should pull the global parameter for density level, not hardcode it to DensityLevelC

Reply:
Thanks!

Reply:
fixed